blocks/library/tweet/index.js

```diff
@@ -0,0 +1,97 @@
+/**
+ * Internal dependencies
+ */
+import { registerBlock, query } from '../../api';
+import Sandbox from '../../../components/sandbox';
+import Button from '../../../components/button';
+import Placeholder from '../../../components/placeholder';
+
+const { prop } = query;
+
+registerBlock( 'core/tweet', {
+	title: wp.i18n.__( 'Tweet' ),
+	icon: 'twitter',
+
+	category: 'social',
+
+	attributes: {
+		url: prop( '*', 'innerHTML' ), // our html is just a div with the url in, for WP's oembed to process
+	},
+
+	edit: class extends wp.element.Component {
+		constructor() {
+			super( ...arguments );
+			this.fetchTweet = this.fetchTweet.bind( this );
+			this.state = {
+				url: this.props.attributes.url,
+				html: '',
+				error: false,
+				fetching: false,
+			};
+		}
+		doFetch( url, setAttributes, setState ) {
+			setState( { fetching: true, error: false } );
+			jQuery.ajax( {
+				type: 'GET',
+				dataType: 'jsonp',
+				data: {},
+				timeout: 5000,
```

Comment:
But is the issue of wanting to avoid waiting a long time for requests specific to this block? I'm curious if and what the default timeout is, or if this should be applied globally through [`$.ajaxSetup`](http://api.jquery.com/jQuery.ajaxSetup/), or whether there's any consensus in WordPress JavaScript that we even need impose a timeout.

More a consistency question than anything.